### ISSUE TITLE
Qube Manager can't open settings tab when /etc/services become UTF-8 non-ascii file

### ISSUE THREAD

**Issue by jpalczewski** [QubesOS/qubes-issues#4475]
### Qubes OS version:

R4.0

### Affected component(s): 

- qube setting page in Qube Manager
- all components which parses /etc/service

---

### Steps to reproduce the behavior:

1. Add non-ascii character in /etc/services(f.e. something above range(128) in comment sections)
2. pre-test: iconv should fail trying to convert the file from UTF-8 to ascii
3. Open settings window for any possible qube

### Expected behavior:

Settings window appears as usual.

### Actual behavior:

`Houston, we have a problem` error window appears, blaming https://github.com/QubesOS/qubes-manager/blob/7a5c304c1e84190c782eeb0065556621a08f4cd9/qubesmanager/firewall.py#L151 and generic notes about badly parsing UTF-8 related chars as ASCII. Qube Manager doesn't dies after that, but refuses to open it on the other hand.

### General notes:

Unfortunately, I wasn't fuzzing qubes manager, only installed KDE5(I blame some dependency, but don't know fedora-foo magic to find it out which package was responsible) which added line with `accuracer 12007/tcp #Accuracer Database System` that messed things up.  

---

**Comment by unman** [comment]
@jpalczewski  I dont seem to be able to reproduce this.
My KDE4 system has an entry for `accuracer 12007/tcp #Accuracer Database System` , and the settings windows open fine.
I've tried inserting a variety of UTF-8 characters in to /etc/services, and haven't yet been able to trigger the bug as reported. Can you say exactly what characters, and location, trigger the bug for you?

**Comment by jpalczewski** [comment]
@unman f.e.
`echo -e "weird 50123/tcp # \xe2\x98\xa0" >> /etc/services ` works as PoC. 
Exactly error says: "UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position ...: ordinal not in range(128)."

**Comment by unman** [comment]
@jpalczewski That doesn't do it for me :-(
Qube manager opens properly, and Settings window opens as expected.
Could it be language?
What version Qube Manager do you have?

**Comment by jpalczewski** [comment]
@unman I have got `pl_PL.UTF-8` recently, but changed it to `c.utf8`(there's were some weird things in `qubes-dom0-update` going on, seems like that tool is output-text dependent and throws some warnings if localized strings appears - it qualifies as another bug, but much less critical - updater works despite warnings AFAIR). 

It's hard to say which version of Qube Manager it is(`qubes-qube-manager -v` won't pass version info, there's no package in "Version Information" about package named in that way either - but it's latest available one(there's no updates available at the moment), but when I looked into `/usr/bin/qubes-qube-manager` ).

I wouldn't blame language in this particular case, it's rather typical "oh-no-unicode" error while parser is looking for ASCII only, where there's no try/catch in function linked above to minimize damage(f.e. just pass one line that fails completely). 
 
If it works in your case that's weird - maybe there is a sideway to suggest encoding for `open('/file', 'r')` function in python3?
